src/lexer.l

```diff
@@ -104,6 +103,8 @@ bpftrace|perf {
 "~"                     { return Parser::make_BNOT(loc); }
 "."                     { return Parser::make_DOT(loc); }
 "->"                    { return Parser::make_PTR(loc); }
+"$"[0-9]+               { return Parser::make_PARAM(yytext, loc); }
```

Comment:
This will match `0000001`, for example

Maybe `"$"[0-9][1-9]*` is better?

Reply:
Hm, what I have is consistent with the current code and `001` is arguably as valid an integer as `1` so I think it'll just be a matter of preference. I'm fine to change to `"$"[1-9][0-9]*` if that's the consensus.